packages/esm-ward-app/src/ward-workspace/admit-patient-form-workspace/admit-patient-form-workspace.workspace.tsx

```diff
@@ -0,0 +1,174 @@
+import { useAdmissionLocation } from '../../hooks/useAdmissionLocation';
+import useWardLocation from '../../hooks/useWardLocation';
+import { useTranslation } from 'react-i18next';
+import { filterBeds } from '../../ward-view/ward-view.resource';
+import type { BedLayout } from '../../types';
+import { z } from 'zod';
+import { Controller, useForm } from 'react-hook-form';
+import { zodResolver } from '@hookform/resolvers/zod';
+import { Form, Row, Column, DropdownSkeleton, Dropdown, ButtonSet, Button } from '@carbon/react';
+import { InlineNotification } from '@carbon/react';
+import { assignPatientToBed, createEncounter } from '../../ward.resource';
+import type { WardConfigObject } from '../../config-schema';
+import { useInpatientRequest } from '../../hooks/useInpatientRequest';
+
+interface AdmitPatientFormWorkspaceProps extends DefaultWorkspaceProps {
+  patient: Patient;
+}
+
+const AdmitPatientFormWorkspace: React.FC<AdmitPatientFormWorkspaceProps> = ({
+  patient,
+  closeWorkspace,
+  closeWorkspaceWithSavedChanges,
+  promptBeforeClosing,
+}) => {
+  const { t } = useTranslation();
+  const { location } = useWardLocation();
+  const [isSubmitting, setIsSubmitting] = useState(false);
+  const { mutate: mutateInpatientRequest } = useInpatientRequest();
+  const { admissionEncounterTypeUuid } = useConfig<WardConfigObject>();
+  const [showErrorNotifications, setShowErrorNotifications] = useState(false);
+  const { isLoading, admissionLocation, mutate: mutateAdmissionLocation } = useAdmissionLocation();
+  const beds = useMemo(() => (isLoading ? [] : filterBeds(admissionLocation)), [admissionLocation]);
+  const getBedRepresentation = useCallback((bedLayout: BedLayout) => {
+    const bedNumber = bedLayout.bedNumber;
+    const patients =
+      bedLayout.patients.length === 0
+        ? [t('emptyBed', 'Empty')]
+        : bedLayout.patients.map((patient) => patient?.person?.preferredName?.display);
+    return [bedNumber, ...patients].join(' · ');
+  }, []);
+
+  const zodSchema = useMemo(
+    () =>
+      z.object({
+        bedId: z.number({
+          required_error: t('bedSelectionRequired', 'Please select a bed to move forward'),
+        }),
+      }),
+    [],
+  );
+
+  type FormValues = z.infer<typeof zodSchema>;
+
+  const {
+    control,
+    formState: { errors, isDirty },
+    handleSubmit,
+    getValues,
+    watch,
+  } = useForm<FormValues>({
+    resolver: zodResolver(zodSchema),
+  });
+
+  useEffect(() => {
+    promptBeforeClosing(() => isDirty);
+  }, [isDirty]);
+
+  const onSubmit = useCallback(
+    (values: FormValues) => {
+      setShowErrorNotifications(false);
+      setIsSubmitting(true);
+      const bedSelected = beds.find((bed) => bed.bedId === values.bedId);
+      createEncounter(patient.uuid, admissionEncounterTypeUuid, location?.uuid)
+        .then(async (response) => {
+          if (response.ok) {
+            return assignPatientToBed(values.bedId, patient.uuid, response.data.uuid);
```

Comment:
The user may not have been able to select a bed, or may have chosen not to select a bed, in which case we shouldn't make this call.

Reply:
The validation is happening with Zod schema.

Reply:
ah, I missed this one during our discussion... there's a line in the ticket that says "It should not be required to select a bed in order to admit the patient onto the ward."  so I think the bed management field should be optional (and therefore we need to support the case when a user doesn't select a bed).